eth/downloader/downloader.go

```diff
@@ -1812,6 +1813,9 @@ func (d *Downloader) importBlockResults(results []*fetchResult) error {
 			// of the blocks delivered from the downloader, and the indexing will be off.
 			log.Debug("Downloaded item processing failed on sidechain import", "index", index, "err", err)
 		}
+		if errors.Is(err, core.ErrAncestorHasNotBeenVerified) {
+			return err
```

Comment:
Return err will make the peer disconnected, shall we return err here?

Reply:
No, the peer will be disconnected is caused by the line 1819, the error was wrapped with errInvalidChain, and line 408 will detect this err to drop the connection. so in this line, we don't return errors with wrapping errInvalidChain, and the connection wouldn't be disconnected.

Reply:
Great